vignettes/data-transformation.Rmd

```diff
@@ -131,4 +130,200 @@ subject `112` had an event at time `51`, thus `log(51-50)=0`:
+
+
+Note that
+
+  - the variable representing follow-up time $t$ in `func` (here `time`) must
+  match the time variable specified on the left-hand side of the `formula`
+
+  - the variable representing exposure time $t_e$ (here `te1` and `te2`) must be
+  wrapped within `latency()` to tell `as_ped` to calculate $t-t_e$
+
+  - by default, $\mathcal{T}(t)$ is defined as `function(t,te) {t >= te}`,
+  thus for $\mathcal{T}_1$ the lag-lead window must not be explicitly specified.
+  In case you want to use a custom lag-lead window, provide the respective
+  function to the `ll_fun` argument in `func` (see `ll2` in the examples above)
+
+  - `func` makes no difference between partial effects such as $h(t-t_e, z(t_e))$
+  and $h(t-t_e)z(t_e)$ as the necessary data transformation is the same in both
+  cases. Later, however, when fitting the model, a distinction must be made
+
+  - more than one $z$ variable can be provided to `func`, which can be convenient,
+  when multiple covariates should have the same time components and the
+  same lag-lead window
+
+  - multiple `func` terms can be specified, having different exposure times
+  `t_e`, `s_e` and/or different lag-lead windows for different covariates
+  $\mathbf{z}_1$, $\mathbf{z}_2$
+
+  - To tell `func` which of the variables specified is the exposure time $t_e$,
+  the `te_var` argument must be specified within each `func` term. The
+  follow-up time component $t$ will be automatically recognized via the left-hand
+  side of the `formula`.
+
+
+### One time-dependent covariate
+For illustration we use the ICU patients
+data sets `patient` and `daily` provided in the **`pammtools`** package, with
+follow-up time `survhosp` ($t$), exposure time `Study_Day` ($t_e$) and TDCs
+`caloriesPercentage` $\mathbf{z}_1$ and `proteinGproKG` ($\mathbf{z}_2$):
+
+```{r}
+head(patient)
+head(daily)
+```
+
+Below we illustrate the usage of `as_ped` and `func` to obtain covariate
+matrices for the latency matrix of follow up time and exposure time and
+a matrix of the TDC (`caloriesPercentage`). The follow up will be split
+in 30 intervals with interval breaks points `0:30`. By default,
+the exposure time provided to `te_var` will be used as a suffix to the new
+matrix columns to indicate with respect to what exposure the columns had been
+constructed, however, you can override the suffix by specifying the
+`suffix` argument to `func`.
+
+```{r}
+ped <- as_ped(
+  data    = list(patient, daily),
+  formula = Surv(survhosp, PatientDied) ~ . |
+    func(latency(Study_Day), caloriesPercentage, te_var="Study_Day"),
+  cut     = 0:30,
+  id = "CombinedID")
+str(ped, 1)
+```
+
+As you can see, the data is transformed to the PED format as before and
+two additional matrix columns are added, `Study_Day_latency` and
+`caloriesPercentage`.<br>&nbsp;
```

Comment:
When I run this, I get names `Study_Day_` and `caloriesPercentage_`...?
![image](https://user-images.githubusercontent.com/998541/38202218-4804cbcc-369b-11e8-914c-8e356fea47a3.png)

Reply:
Suffix not used because of : 
```r 
 if (is.null(func$suffix) & n_func == 1) {
    suffix <- ""
  } else {
    suffix <- func$suffix
  }
``` 
https://github.com/adibender/pammtools/blob/devel/R/trafo-cumulative.R#L83